framework/src/media/FileInputDataSource.cpp

```diff
@@ -73,19 +73,23 @@ bool FileInputDataSource::open()
 		case AUDIO_TYPE_MP3:
 		case AUDIO_TYPE_AAC:
 			if (!utils::header_parsing(mFp, audioType, &channel, &sampleRate, NULL)) {
-				medvdbg("header parsing failed\n");
+				meddbg("header parsing failed\n");
+				return false;
+			}
+			if (fseek(mFp, 0, SEEK_SET) != 0) {
```

Comment:
is that right?
That function will do it already(header_parsing).
	if (fseek(fp, 0, SEEK_SET) != 0) {
		meddbg("file seek failed error\n");
		return false;
	}

Reply:
in audio_decoder.cpp, we need to parse wave header data, there are other params which header_parsing() has not parsed for decoder.

Reply:
But, you can do the same thing twice.

Reply:
you are right! I didn't see that. fseek(mFp, 0, SEEK_SET) is not necessary here.
Thanks.

Reply:
@jsdosa fixed, thanks.